### PR TITLE
POLIO-187 fix colors

### PR DESCRIPTION
**Pull request by quang-le**
## Changes

- use `backgroundLayer` prop of `MapComponent` for shapes that have no status and are not in scope
- refactor API calls to remove duplicate calls. Some prop drilling was reintroduced, but I didn't want to introduce redux in polio, and a `Context` would have introduced expensive additional map renders. The `useQuery` hooks with `select` options where replaced by hooks transforming and memoizing the data, which is then passed as props
- removed `countryOfSelectedCampaign` and replaced its uses with `country`, which allows the map to center on the country when selecting it
- removed `LoadingSpinner`from `NoFingerMzrk`as it was blocking the whole screen and "duplicating" other existing spinners

https://user-images.githubusercontent.com/38907762/152181116-eb44a3a9-e0e7-4581-b501-593daf95ae8f.mov

### CODE REVIEW

**Review by beygorghor — APPROVED** [review]
As discussed you can use react-query caching here https://react-query.tanstack.com/guides/caching to avoid double call
By default 'staletime 'is set to zero.

Also those pages are quiet complex, it can be a good thing cypress/test it one day.

**Comment by beygorghor** [comment]
Screenshot 2022-02-03 at 12 09 56
Also i have this error in the console

File: plugins/polio/js/src/components/LQAS-IM/LqasImMap.js [hunk 1]
```diff
@@ -56,35 +61,58 @@ export const LqasImMap = ({
...
-    const getShapeStyles = useCallback(
+    const mainLayer = useMemo(() => {
+        if (isEqual(data, {})) return [];
+        if (!selectedCampaign) return [];
+        const determineStatusForDistrict =
+            type === 'lqas' ? lqasDistrictStatus : imDistrictStatus;
+        const scopeIds = findScopeIds(selectedCampaign, campaigns);
+        const hasScope = scopeIds.length > 0;
+        const shapesInScope = hasScope
+            ? shapes.filter(shape => scopeIds.includes(shape.id))
```
[the remark this answers is not in the document]
Comment (quang-le) [reply plugins/polio/js/src/components/LQAS-IM/LqasImMap.js:72]: yes, it's even surprising it works, since `filter`returns an array

**Comment by quang-le** [comment]
> Screenshot 2022-02-03 at 12 09 56
> Also i have this error in the console

Ah, crap. I did convert the value coming from params to string, because it was causing the reverse error in Autocomplete (received a number while expecting a string). I'll look it up, so it doesn't clutter the console

**Pull request merged**